src/test/java/org/jenkinsci/plugins/durabletask/BourneShellScriptTest.java

```diff
@@ -151,4 +176,54 @@ public void smokeTest() throws Exception {
         c.cleanup(ws);
     }
 
+    @Test public void shebang() throws Exception {
+        Controller c = new BourneShellScript("#!/bin/cat\nHello, world!").launch(new EnvVars(), ws, launcher, listener);
+        while (c.exitStatus(ws, launcher) == null) {
+            Thread.sleep(100);
+        }
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        c.writeLog(ws, new TeeOutputStream(baos, System.out));
+        assertEquals(0, c.exitStatus(ws, launcher).intValue());
+        assertThat(baos.toString(), containsString("Hello, world!"));
+        c.cleanup(ws);
+    }
+
+    @Test public void runOnUbuntuDocker() throws Exception {
+        JavaContainer container = dockerUbuntu.get();
+        runOnDocker(new DumbSlave("docker", "/home/test", new SSHLauncher(container.ipBound(22), container.port(22), "test", "test", "", "")));
+    }
+
+    @Test public void runOnCentOSDocker() throws Exception {
+        CentOSFixture container = dockerCentOS.get();
+        runOnDocker(new DumbSlave("docker", "/home/test", new SSHLauncher(container.ipBound(22), container.port(22), "test", "test", "", "")));
+    }
+
+    private void runOnDocker(DumbSlave s) throws Exception {
+        j.jenkins.addNode(s);
+        j.waitOnline(s);
+        FilePath dockerWS = s.getWorkspaceRoot();
+        Launcher dockerLauncher = s.createLauncher(listener);
+        Controller c = new BourneShellScript("echo hello world; sleep 10").launch(new EnvVars(), dockerWS, dockerLauncher, listener);
+        while (c.exitStatus(dockerWS, dockerLauncher) == null) {
+            Thread.sleep(100);
+        }
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        c.writeLog(dockerWS, baos);
+        assertEquals(0, c.exitStatus(dockerWS, dockerLauncher).intValue());
+        assertTrue(baos.toString().contains("hello world"));
+        c.cleanup(dockerWS);
+        do {
+            Thread.sleep(1000);
+            baos = new ByteArrayOutputStream();
+            assertEquals(0, dockerLauncher.launch().cmds("ps", "-e", "-o", "pid,stat,command").stdout(new TeeOutputStream(baos, System.out)).join());
+        } while (baos.toString().contains(" sleep "));
+        assertThat("no zombies running", baos.toString(), not(containsString(" Z ")));
+        s.toComputer().disconnect(new OfflineCause.UserCause(null, null));
+    }
+
+    @Test public void runWithCommandLauncher() throws Exception {
+        assumeTrue("Docker required for this test", new Docker().isAvailable());
+        runOnDocker(new DumbSlave("docker", "/home/jenkins/agent", new SimpleCommandLauncher("docker run -i --rm --name agent jenkinsci/slave:3.7-1 java -jar /usr/share/jenkins/slave.jar")));
```

Comment:
@MarkEWaite points me to tini, which is apparently available as an `--init` option, so I should check whether that _also_ solves the zombie issue without `wait`, and if so, recommend its usage on the image page.